Shorten quick start (#2399)
meilisearch/documentation: learn/getting_started/quick_start.mdx

# Quick start

This quick start walks you through installing Meilisearch, adding documents, and performing your first search.

To follow this tutorial you need:

- A [command line terminal](https://www.learnenough.com/command-line-tutorial#sec-running_a_terminal)
- [cURL](https://curl.se) 
- Optional: one of the official Meilisearch [SDKs](/learn/what_is_meilisearch/sdks) if you want to follow along in a specific programming language

## Setup and installation

First, you need to download and install Meilisearch. This command installs Meilisearch in your local machine:

```bash
# Install Meilisearch
curl -L https://install.meilisearch.com | sh
```

The rest of this guide assumes you are using Meilisearch locally, but you may also use Meilisearch over a cloud service such as [Meilisearch Cloud](https://www.meilisearch.com/pricing). 

Learn more about other installation options in the [installation guide](/learn/getting_started/installation).

### Running Meilisearch

Next, launch Meilisearch by running the following command in your terminal:

```bash
# Launch Meilisearch

> ./meilisearch --master-key="aSampleMasterKey"

Reviewer: ```suggestion
./meilisearch --master-key="MASTER_KEY"
```

```

> You should see something a bit like this in response:

Reviewer: A bit sounds like users may get different responses. Wouldn't "you should see the following response" or something be better?

Author: I sort of want to be imprecise here because the exact message may look different due to user setup (the Meilisearch ASCII art may get messed up depending on your terminal size) or to minor changes made by the engine team (for example, users nowadays see `Config file path` before `Database path`.

Server listening on: "localhost:7700"
```

You now have a Meilisearch instance running in your terminal window. Keep this window open for the rest of this tutorial.

This command uses the `--master-key` configuration option to secure Meilisearch. This tutorial uses `aSampleMasterKey` as a master key, but you may change it to any alphanumeric string with 16 or more characters.

> Setting a master key is optional but strongly recommended in development environments. Master keys are mandatory in production environments. To learn more about securing Meilisearch, refer to the [API keys guide](/learn/security/master_api_keys).

Reviewer 2: ```suggestion
To learn more about securing Meilisearch, refer to the [master and API keys guide](/learn/security/master_api_keys).
```

## Add documents

In this quick start, you will search through a collection of movies.

To follow along, first click this link to download the file: <ButtonLink as="a" id="downloadMovie" href="/movies.json" download="movies.json">movies.json</ButtonLink>. Then, move the downloaded file into your working directory.

<Capsule intent="note">
Meilisearch accepts data in JSON, NDJSON, and CSV formats.
</Capsule>

Run the following command, opening a new terminal window if you are following this tutorial using `cURL`:

<CodeSamples id="getting_started_add_documents_md" />

Meilisearch stores data in the form of discrete records, called [documents](/learn/core_concepts/documents). Each document is composed of multiple fields, which are pairs of one attribute and one value:

```json
{
  "id": 0,
  "attribute": "value"
}
```

Documents are grouped into collections, called [indexes](/learn/core_concepts/indexes).

> Every index must have a [primary key](/learn/core_concepts/primary_key#primary-field), an attribute shared across all documents in that index. If you try adding documents to an index and even a single one is missing the primary key, none of the documents will be stored.

Reviewer 2: WDYT about putting all of these into a Note capsule? (could be titled "Documents, indexes, and the primary key"

Reasoning: the info-dumping here is important, but it also really breaks up the flow of the guide, with the paragraph on line 83 referring to a "previous command" on line 68, which may also have been run a long time ago depending on whether the reader follows links to learn more about documents, indexes, and/or primary keys. Would be better to group them into more of a parenthetical.

<Capsule intent="note">
The above command uses the `--master-key` configuration option to secure Meilisearch. Setting a master key is optional but strongly recommended in development environments. Master keys are mandatory in production environments.
</Capsule>

To learn more about securing Meilisearch, refer to the [master and API keys guide](/learn/security/master_api_keys).

## Add documents

In this quick start, you will search through a collection of movies.

To follow along, first click this link to download the file: <ButtonLink as="a" id="downloadMovie" href="/movies.json" download="movies.json">movies.json</ButtonLink>. Then, move the downloaded file into your working directory.

<Capsule intent="note">
Meilisearch accepts data in JSON, NDJSON, and CSV formats.
</Capsule>

Open a new terminal window and run the following command:

<CodeSamples id="getting_started_add_documents_md" />

Meilisearch stores data in the form of discrete records, called [documents](/learn/core_concepts/documents). Each document is composed of multiple fields, which are pairs of one attribute and one value:

```json
{
  "id": 0,

>   "attribute": "value"

Author: I guess you're right. I thought it could be good to show the id attribute, but now you mention it `id` might distract users from the main point we're trying to illustrate here.

}
```

By default, Meilisearch only returns the first 20 results for a search query. You can changed this using the [`limit` parameter](/reference/api/search#limit).

## What's next?

> You now know all the basics: how to install Meilisearch, create an index, add documents, check the status of an asynchronous task, and search.

Reviewer 2: ```suggestion
You now know how to install Meilisearch, create an index, add documents, check the status of an asynchronous task, and make a search request.
```

If you want to try out Meilisearch in a web interface, take a look at the [search preview](/learn/getting_started/search_preview). You can also build your own [front-end integration](/learn/getting_started/front_end_integration).

For a guided overview of some of Meilisearch's main features, continue to the [Meilisearch 101](/learn/getting_started/filtering_and_sorting).

> For a more advanced approach, consult the [API reference](/reference/api/overview).

Author: I chose to keep these separate for easier scanning so each ending paragraph provides a different next step. I'm afraid bundling them all together might make a significant block of text and risk an "ain't nobody got time for that" reaction.

}
```

By default, Meilisearch only returns the first 20 results for a search query. You can change this using the [`limit` parameter](/reference/api/search#limit).

## What's next?

You now know how to install Meilisearch, create an index, add documents, check the status of an asynchronous task, and make a search request.

> If you'd like to search through the documents you just added using a clean browser interface rather than the terminal, you can do so with [our built-in search preview](/learn/getting_started/search_preview). You can also [learn how to quickly build a front-end interface](/learn/getting_started/front_end_integration) of your own.

Reviewer: ```suggestion
If you'd like to search through the documents you just added using a clean browser interface rather than the terminal, you can do so with [our built-in search preview](/learn/getting_started/search_preview). You can also [learn how to quickly build a front-end interface](/learn/front_end/front_end_integration) of your own.
```
